luigi/contrib/bigquery.py

```diff
@@ -668,6 +684,91 @@ class ExternalBigQueryTask(MixinBigQueryBulkComplete, luigi.ExternalTask):
     pass
 
 
+class BigQueryExtractTask(luigi.Task):
+    """
+    Extracts (unloads) a table from BigQuery to GCS.
+
+    This tasks requires the input to be exactly one BigQueryTarget while the
+    output should be one or more GCSTargets from luigi.contrib.gcs depening on
+    the use of destinationUris property.
+    """
+    @property
+    def destination_uris(self):
+        """
+        The fully-qualified URIs that point to your data in Google Cloud
+        Storage. Each URI can contain one '*' wildcard character and it must
+        come after the 'bucket' name.
+
+        Wildcarded destinationUris in GCSQueryTarget might not be resolved
+        correctly and result in incomplete data. If a GCSQueryTarget is used to
+        pass wildcarded destinationUris be sure to overwrite this property to
+        suppress the warning.
+        """
+        return [x.path for x in luigi.task.flatten(self.output())]
+
+    @property
+    def print_header(self):
+        """Whether to print the header or not."""
+        return PrintHeader.TRUE
+
+    @property
+    def field_delimiter(self):
+        """
+        The separator for fields in a CSV file. The separator can be any
+        ISO-8859-1 single-byte character.
+        """
+        return FieldDelimiter.COMMA
+
+    @property
+    def destination_format(self):
+        """
+        The destination format to use (see :py:class:`DestinationFormat`).
+        """
+        return DestinationFormat.CSV
+
+    @property
+    def compression(self):
+        """Whether to use compression."""
+        return Compression.NONE
+
+    def run(self):
+        input = luigi.task.flatten(self.input())[0]
+        assert (
+            isinstance(input, BigQueryTarget) or
+            (len(input) == 1 and isinstance(input[0], BigQueryTarget))), \
+            'Input must be exactly one BigQueryTarget, not %s' % (input)
+        bq_client = input.client
+
+        destination_uris = self.destination_uris
+        assert all(x.startswith('gs://') for x in destination_uris)
+
+        logger.info('Launching Extract Job')
+        logger.info('Extract source: %s', input)
+        logger.info('Extract destination: %s', destination_uris)
+
+        job = {
+            'configuration': {
+                'extract': {
+                    'sourceTable': {
+                        'projectId': input.table.project_id,
+                        'datasetId': input.table.dataset_id,
+                        'tableId': input.table.table_id
+                    },
+                    'destinationUris': destination_uris,
+                    'printHeader': self.print_header,
```

Comment:
Good catch! I never used with BigQuery with anything other than `csv` and am therefore not aware of the behavior. I know from other cases than superfluous job parameter are ignored so I would _assume_ same is true for `printHeader` and `fieldDelimiter`.

Regardless it would be more consistent to set `printHeader` and `fieldDelimiter` conditionally. I will add as soon as find some time.